### PR TITLE
Methods to write and read jason

### PR DESCRIPTION
**Pull request by kmusial1988**
Methods to write and read jason

### CODE REVIEW

File: src/main/java/pl/sda/weather/controllers/LocationController.java [hunk 1]
```diff
@@ -2,57 +2,61 @@
...
+import java.util.Map;
 import java.util.Scanner;
 import java.util.UUID;
 
 public class LocationController {
-    ILocationService locationService = new LocationServiceImpl();
+
+    private static final ILocationService locationService = new LocationServiceImpl();
 
     private static final Scanner scanner = new Scanner(System.in);
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/controllers/LocationController.java:20]: classes related with View, should be in View class (not in Controller).

File: src/main/java/pl/sda/weather/controllers/LocationController.java [hunk 2]
```diff
@@ -2,57 +2,61 @@
...
 
-    private static LocationValidation validation = new LocationValidation();
+    private static LocationValidator validation = new LocationValidator();
+
+    ReadWeathersImpl readWeathers = new ReadWeathersImpl();
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/controllers/LocationController.java]: should be private

File: src/main/java/pl/sda/weather/controllers/LocationController.java [hunk 3]
```diff
@@ -2,57 +2,61 @@
...
-//
-//           longitudeAndLatitude = "(latitude: " + longitudeS + " -> S; " + longitudeN + " -> N; longitude: " + latitudeW + " -> W; " + latitudeE + " -> E)";
-//
-//
-//       }while (latitudeE.equals("") || latitudeW.equals("") || longitudeN.equals("") || longitudeS.equals(""));
+        String longitude;
+        String latitude;
+
+        do {
+            System.out.println("Write city latitude in format (000.00) : ");
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/controllers/LocationController.java:35]: same as before -> classes related with View, should be in View class (not in Controller).

File: src/main/java/pl/sda/weather/repository/ILocationRepository.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.List;
+
+public interface ILocationRepository {
+
+
+    void addLocationModelToTxtDB(LocationModel locationModel);
+
+    void addLocationModelJsonToDB(LocationModel locationModel);
+
+    List<LocationModel> getLocationModelFromFileTxt();
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/repository/ILocationRepository.java]: idea behind interface is to have only one method :)

```
List<LocationModel> getLocationModelFromFileTxt();
List<LocationModel> getLocationModelFromFileJson();
```

It's the same method and should have two implementations in LocationTxtRepository and LocationJsonRepository

File: src/main/resources/location/locations.txt [hunk 5]
```diff
@@ -1,5 +1,2 @@
-4b5f417d-b193-4ce8-b63b-f0983e087501,(latitude: SS -> S; SS -> N; longitude: SS -> W; SS -> E),KRK,MAŁ,POL
-4b5f417d-b193-4ce8-b63b-f0983e087501,(latitude: SS -> S; SS -> N; longitude: SS -> W; SS -> E),KRKGGG,MAŁGGG,POL
-6e9c7139-45f4-4ada-8307-af44961af1eb,(latitude: 111 -> S; 11 -> N; longitude: 22 -> W; 44 -> E),krrr,mllll,polll
-a8ab2543-1f3c-409d-9000-da06bf3bcb6c,(latitude: 1 -> S; 1 -> N; longitude: 1 -> W; 1 -> E),1,1,1
-662da5e6-31ef-4872-82cb-4915a07f37b9,lok,d,User don't add regional,d
+acd07c55-09d9-406d-b921-6865410a4764,latitude: 002.22; longitude: 222.66,Krakow,Małopolska,Polska
```
Comment (MichalOrlowski) [inline src/main/resources/location/locations.txt]: in inital phase this file should be empty, right?

File: src/main/java/pl/sda/weather/validation/LocationValidator.java [hunk 6]
```diff
@@ -3,24 +3,21 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-public class LocationValidation {
+public class LocationValidator {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/validation/LocationValidator.java:6]: Waiting for tests :)

File: src/main/java/pl/sda/weather/service/impl/ReadWeathersImpl.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Stream;
+
+public class ReadWeathersImpl implements ReadWeathers {
+
+    @Override
+    public Map<String, Weather> mapWeather() {
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/service/impl/ReadWeathersImpl.java]: still waiting for tests

File: src/main/java/pl/sda/weather/repository/impl/LocationRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class LocationRepositoryImpl implements ILocationRepository {
+
+    private static final String FILE_LOCATION = "src/main/resources/location/locations.txt";
```
Comment (MichalOrlowski) [inline src/main/java/pl/sda/weather/repository/impl/LocationRepositoryImpl.java]: would be nice to have test for that also -> this FILE_LOCATION also should be move from constant to constructor

**Review by Wielopol — APPROVED** [review]

**Pull request merged**